cmake/kokkos_arch.cmake

```diff
@@ -448,6 +448,9 @@ UNSET(_HOST_PARALLEL)
 
 IF(KOKKOS_ENABLE_PTHREAD)
   SET(KOKKOS_ENABLE_THREADS ON)
+  ARCH_FLAGS(
```

Comment:
@jjwilke is this the right place to add it?

Reply:
It's probably not an architecture flag. I would probably put it in `kokkos_enable_devices.cmake` around line 26 where there's already a pthread if-block.

I started the compiler options as global variables... so leave them that way for now. They can probably become regular variables. Anyhow, the relevant code is:

````
  LIST(APPEND KOKKOS_COMPILE_OPTIONS -pthread)
  LIST(APPEND KOKKOS_LINK_OPTIONS -pthread)
  GLOBAL_SET(KOKKOS_COMPILE_OPTIONS ${KOKKOS_COMPILE_OPTIONS})
  GLOBAL_SET(KOKKOS_LINK_OPTIONS ${KOKKOS_LINK_OPTIONS})
````

I wasn't sure how the Trilinos stuff was going to shake out so I made several things globals that weren't necessary to be globals. If we convert these to regular variables, you can just append:

````
  LIST(APPEND KOKKOS_COMPILE_OPTIONS -pthread)
  LIST(APPEND KOKKOS_LINK_OPTIONS -pthread)
````

Reply:
@jjwilke can you make a concrete counter proposal for this?